server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java

```diff
@@ -171,6 +172,37 @@ private void setDefault(DbSession session, QualityProfileDto profile) {
     db.qualityProfileDao().update(session, profile.setDefault(true));
   }
 
+  public QualityProfileDto find(QProfileRef ref) {
```

Comment:
yes, you're right, but it's very convenient for callers like ChangelogAction that do not need to inject DbClient. Still I agree that it's risky to nest such calls when a DbSession is already open.